l10n_ch_pain_base/models/account_payment_order.py

```diff
@@ -68,7 +68,8 @@ def generate_party_agent(
             if bank_line.local_instrument == 'CH01':
                 # Don't set the creditor agent on BVR/CH01 payments
                 return True
-            elif not partner_bank.bank_bic:
+            elif not partner_bank.bank_bic and \
+                    self.payment_method_id.bic_required:
```

Comment:
The rule must be about using an IBAN or QR-IBAN.

If IBAN -> we don't need a BIC (it can be deduced from IBAN)

```suggestion
            elif not partner_bank.bank_bic and not partner_bank.type == 'iban':
```